pkg/fleet/installer/setup/djm/databricks.go

```diff
@@ -220,6 +223,66 @@ func setClearHostTag(s *common.Setup, tagKey, value string) {
 	s.Span.SetTag("host_tag_value."+tagKey, value)
 }
 
+func setupGPUIntegration(s *common.Setup) {
+	gpuEnabled := os.Getenv("GPU_MONITORING_ENABLED")
+
+	if gpuEnabled != "1" && gpuEnabled != "true" {
+		return
+	}
+
+	s.Out.WriteString("GPU monitoring enabled via GPU_MONITORING_ENABLED environment variable\n")
+
+	s.Config.DatadogYAML.CollectGPUTags = true
+	s.Config.DatadogYAML.EnableNVMLDetection = true
+
+	if s.Config.SystemProbeYAML == nil {
+		s.Config.SystemProbeYAML = &config.SystemProbeConfig{}
+	}
+	s.Config.SystemProbeYAML.GPUMonitoringConfig = config.GPUMonitoringConfig{
+		Enabled: true,
+	}
+
+	s.Span.SetTag("gpu_monitoring_enabled", "true")
+
+	// NVML is not initialized when the databricks init script executes.
+	// This causes the GPU integration to fail initialisation, we must retry after
+	// a timeout so the NVML checks are passing and the GPU integration starts polling metrics.
+	scheduleDelayedAgentRestart(s, 30*time.Second)
+}
+
+// scheduleDelayedAgentRestart schedules an agent restart after the specified delay
+func scheduleDelayedAgentRestart(s *common.Setup, delay time.Duration) {
```

Comment:
Some more details about this: I initially tried with a goroutine but since the init script terminates, it also kills the goroutine and the delayed restart doesn't happen.
In the end the best solution i've found is this duct-taped bash script. We work in the `/tmp` folder to avoid permissions issues.